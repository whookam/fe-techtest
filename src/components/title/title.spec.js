import React from 'react';
import TestRenderer from 'react-test-renderer';
import { render } from '@testing-library/react';
import 'jest-styled-components';

import Title from './index';

describe('Title Component', () => {
  const { getByTestId } = render(<Title title="Which? title component" />);

  it('displays the correct content when given a title to render', () => {
    const renderedTitle = getByTestId('title');
    expect(renderedTitle.innerHTML).toEqual('Which? title component');
  });

  test("Render snapshot test of the title component", () => {
    const renderedTitle = TestRenderer.create(<Title title="Snapshot title" />).toJSON();
    expect(renderedTitle).toMatchSnapshot();
  });
});
