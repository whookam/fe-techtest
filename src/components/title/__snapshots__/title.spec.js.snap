// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Title Component Render snapshot test of the title component 1`] = `
.c0 {
  color: #333;
  font-size: 32px;
  font-family: 'Helvetica Neue, Segoe, Verdana, sans-serif';
}

<h1
  className="c0"
  data-testid="title"
>
  Snapshot title
</h1>
`;
